### PR TITLE
Add tests for sycl::build

### PR DESCRIPTION
**Pull request by vasilytric**
These tests do these verifications:
- if compilation and linkage is possible all kernels without optional features should be represented in the result bundle
- if compilation and  linkage is possible only compatible kernels should be represented in the result bundle
- kernels, that provided to the kernel bundle should be invokable (if such kernel compatible with current device)
- exception was thrown in case device not from provided kernel bundle devices
- exception was thrown for zero devices passed
- provided context is equal to the obtained bundle context
- associated devices are equal to devices from obtained kernel bundle
- `property_list` existence and empty `property_list` can be passed or omitted

______________________________________
This PR depend on these PRs:
- https://github.com/KhronosGroup/SYCL-CTS/pull/218
- https://github.com/KhronosGroup/SYCL-CTS/pull/220
- https://github.com/KhronosGroup/SYCL-CTS/pull/226

### CODE REVIEW

**Review by ProGTX — APPROVED** [review]

File: tests/kernel_bundle/sycl_build_zero_device.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  void get_info(test_base::info &out) const override {
+    set_test_info(out, TOSTRING(TEST_NAME), TEST_FILE);
+  }
+
+  /** execute the test
+   */
+  void run(util::logger &log) override {
+    auto q = util::get_cts_object::queue();
+
+    std::vector<sycl::device> zero_device{};
```
Comment (keryell) [inline tests/kernel_bundle/sycl_build_zero_device.cpp:39]:
```suggestion
    std::vector<sycl::device> zero_device;
```

**Review by keryell — CHANGES_REQUESTED** [review]
Good.

File: tests/kernel_bundle/sycl_build_zero_device.cpp (hunk 1 above)
Comment (vasilytric) [reply tests/kernel_bundle/sycl_build_zero_device.cpp]: I've applied this comment in this commit: https://github.com/KhronosGroup/SYCL-CTS/pull/228/commits/546b2aa975eb879a6330732e1b60ab310626ef7b

**Review by keryell — APPROVED** [review]
Thanks.

**Comment by vasilytric** [comment]
I am going to merge `SYCL-2020` branch into this branch after PR#226 will be merged because of the fact that this PR is depend on PR#226

**Comment by vasilytric** [comment]
@bader All verifications have passed. Would you mind to merge this PR?

**Pull request merged**